compiler/rustc_lint/src/internal.rs

```diff
@@ -67,6 +71,8 @@ fn typeck_results_of_method_fn<'tcx>(
 }
 
 declare_tool_lint! {
+    /// The `potential_query_instability` lint detects use of methods which can lead to
+    /// potential query instability.
```

Comment:
Would be nice to add examples of which methods are detected. (Or really, just "iterating over a HashMap").

Maybe also a link to something with more info would je nice (maybe the rustc dev guide on incremental?)